### PR TITLE
Publishers found in v4 channel responses should have verified checkmark

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/10729

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Open browser with clean profile and `--rewards=staging=true`.
- Enable rewards.
- Dismiss grant notification in the rewards panel.
- Navigate to "duckduckgo.com".
  - Verify that checkmark is displayed on rewards badge.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by zenparsing** [comment]
CI failed on unrelated unit tests on Android, and known macOS "test-install" issue.

File: components/brave_rewards/browser/test/common/rewards_browsertest_response.cc [hunk 1]
```diff
@@ -331,43 +332,6 @@ void RewardsBrowserTestResponse::Get(
     }
   }
 
-  if (URLMatches(
```
Comment (zenparsing) [inline components/brave_rewards/browser/test/common/rewards_browsertest_response.cc:334]: Old v3 test data that wasn't previously removed.

File: components/brave_rewards/browser/test/rewards_publisher_browsertest.cc [hunk 2]
```diff
@@ -147,6 +147,18 @@ IN_PROC_BROWSER_TEST_F(RewardsPublisherBrowserTest, VisitUnverifiedPublisher) {
       false);
 }
 
+// Registered publishers without a wallet address are displayed as verified
+IN_PROC_BROWSER_TEST_F(RewardsPublisherBrowserTest, VisitRegisteredPublisher) {
+  rewards_browsertest_helper::EnableRewards(browser());
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/test/rewards_publisher_browsertest.cc:152]: please use enable via code

Comment (zenparsing) [reply components/brave_rewards/browser/test/rewards_publisher_browsertest.cc:152]: The following call `VisitPublisher` expects the current tab to have the rewards page loaded, so using `EnableViaCode` will require us to load that page and wait first.

For consistency with the other `Visit*Publisher` tests, would it be OK to leave this as-is?

Comment (NejcZdovc) [reply components/brave_rewards/browser/test/rewards_publisher_browsertest.cc:152]: sound good

**Comment by LaurenWags** [comment]
Verification completed using

```
Brave | 1.13.24 Chromium: 84.0.4147.89 (Official Build) nightly (64-bit)
-- | --
Revision | 19abfe7bcba9318a0b2a6bc6634a67fc834aa592-refs/branch-heads/4147@{#852}
OS | macOS Version 10.14.6 (Build 18G3020)
```

* Verified test plan from PR

* Reproduced the issue using `1.13.10`:


* However, on upgrade to `1.13.24`, DDG still did not display the verified icon:
1 13 24 upgrade

* Refreshing status from the panel fixed this as the incorrect publisher info was cached in the DB. Confirmed with @zenparsing this is expected.

* Verified clean profile case works for `1.13.24`:
Screen Shot 2020-07-17 at 8 20 40 AM

**Pull request merged**